src/BuiltInTools/DotNetDeltaApplier/HotReloadAgent.cs

```diff
@@ -213,6 +215,29 @@ public void ApplyDeltas(IReadOnlyList<UpdateDelta> deltas)
             }
         }
 
+        public void ApplyDeltas(Assembly assembly, IReadOnlyList<UpdateDelta> deltas)
+        {
+            try
+            {
+                // Defer discovering the receiving deltas until the first hot reload delta.
+                // This should give enough opportunity for AppDomain.GetAssemblies() to be sufficiently populated.
+                _handlerActions ??= GetMetadataUpdateHandlerActions();
+                var handlerActions = _handlerActions;
+
+                for (var i = 0; i < deltas.Count; i++)
```

Comment:
This may as well be a foreach now right?

Reply:
Are you advocating for more allocations?

Reply:
Yep